### ISSUE TITLE
Emoji images scrambled

### ISSUE THREAD

**Issue by GrumpyOldTroll** [zulip/zulip-desktop#476]
---
- **Operating System**:
  - [ * ] macOS: Sierra: 10.12.6 (16G1314)
- **Clear steps to reproduce the issue**:
  - unknown. started today, I took no unusual action, and have never seen this before. Persists through restart or refresh
- **Relevant error messages and/or screenshots**:
![screen shot 2018-04-19 at 1 40 16 pm](https://user-images.githubusercontent.com/361578/39017176-6493a44a-43d7-11e8-9c8f-9982e45b9e32.png)
![screen shot 2018-04-19 at 1 39 57 pm](https://user-images.githubusercontent.com/361578/39017204-73d70dc0-43d7-11e8-85ef-4b7d0b9156b7.png)
- extra info:
  - I do not currently see the same issue when using the web ui and chrome:
![screen shot 2018-04-19 at 1 47 49 pm](https://user-images.githubusercontent.com/361578/39017498-4eea2640-43d8-11e8-8983-2cbca90e4cfc.png)
  - client version on about page is v1.9.0

**Comment by timabbott** [comment]
Adding @HarshitOnGitHub to this thread since he may be able to help debug.  But I kinda suspect this is some sort of browser caching bug (since you folks recently upgraded to the release where we moved to the 64x64 emoji sprite sheet).

**Comment by HarshitOnGitHub** [comment]
I am also sure this is some sort of caching related issue. I get this scrambled view whenever I change some emoji infra code but older spritesheets from the cache gets used. A hard reload always fixes the issue for me.

**Comment by akashnimare** [comment]
@GrumpyOldTroll does `CMD+SHIFT+R` or restarting the app fix the issue?

**Comment by abhigyank** [comment]
@GrumpyOldTroll  You can also try `Reset App Settings` in settings and see if restarting everything afresh works.

**Comment by GrumpyOldTroll** [comment]
None of these fixed it. Something else I can try?

**Comment by timabbott** [comment]
I'm out of plausible ideas for figuring out what might be special about your environment.  But you can try switching emoji styles in "display settings" and see if that helps?

**Comment by abhigyank** [comment]
@GrumpyOldTroll Can you try this ->
* View Menu (on the top) -> `Toggle DevTools For Active Tab`
* In the Developer Tools console, Go to `Applications` Tab. 
* On the left sidebard, go to `Clear Storage` .
* Click the `clear site data` button in the main view.

Then you'd need to relogin on the site. See if this works.

**Comment by GrumpyOldTroll** [comment]
That didn't work either, but it was a very interesting view, so I poked around a little after trying it, and I see a likely source of the difference between the browser and the app:

In-browser, https://<our server>/static/generated/emoji/google_sprite.css looks very different from the same url for the app. Specifically, the browser references sheet_google_64.png, and the app references sheet_google_32.png, and they have different emoji locations, but the /static/generated/emoji/sheet_google_32.png and /static/generated/emoji/sheet_google_64.png objects themselves look the same to me when I fetch them in browser (though they have different file sizes, so they're not actually identical).

From browser:
```
div.emoji,
span.emoji
{
    display: inline-block;
    background-image: url('sheet_google_64.png');
    -webkit-background-size: 4900%;
    -moz-background-size: 4900%;
    background-size: 4900%;
    background-repeat: no-repeat;

    /* Hide the text. */
    text-indent: 100%;
    white-space: nowrap;
    overflow: hidden;
}

.emoji-00a9 {
    background-position: 0.0% 0.0%;
}
.emoji-00ae {
    background-position: 0.0% 2.0833333333333335%;
}
.emoji-203c {
    background-position: 0.0% 4.166666666666667%;
}
...many more...
```

From app:
```
div.emoji,
span.emoji
{
    display: inline-block;
    background-image: url('sheet_google_32.png');
    -webkit-background-size: 4100%;
    -moz-background-size: 4100%;
    background-size: 4100%;
    background-repeat: no-repeat;

    /* Hide the text. */
    text-indent: 100%;
    white-space: nowrap;
    overflow: hidden;
}

.emoji-00a9 {
    background-position: 0.0% 0.0%;
}
.emoji-00ae {
    background-position: 0.0% 2.5%;
}
.emoji-203c {
    background-position: 0.0% 5.0%;
}
...many more...
```

This mismatch also would possibly seem to explain why the emojis don't seem to line up on proper emoji image boundaries also, if the google_sprite.css is not matched up with the correct sheet_google_xx.png?

Does that analysis help suggest anything else to try?
[edit: formatting for pasted css contents]

**Comment by timabbott** [comment]
Interesting; well that confirms that you're still getting (cached, I assume) CSS from the previous version of Zulip.  The puzzle is why.  Can you try using the developer tools network tab to see how your browser is loading that CSS file (whether it's cached or being downloaded from the server)?  

If it's being downloaded from the server, one possibility that this makes me think about is whether your organization has multiple servers running different versions, and your client is somehow connecting to the wrong one (i.e. basically a DNS issue).

**Comment by GrumpyOldTroll** [comment]
I'm definitely re-fetching the google_sprite.css, I saw a 200 in the network view of developer tools on reload, and the numbers seem not in sync with the number of emoji columns in the sheet, so yeah.

The DNS/unpurged thing occurred to me too, so I checked the IP of somebody who's not experiencing this problem, and it's not as straightforward as cache being out of date on that machine. (Also: I definitely reliably see a different google_sprite.css from the browser and the app, even though they're both running on the same machine and getting the same DNS results.)

I tried to see the headers I'm sending in the request, but here I got stuck from what seems to be another different bug. I'm now on v2.0.0 and in this build, when I have the devtools for active tab open and I do a refresh, the app hangs and I have to restart, so I can't see what the headers are from the network tab of devtools. During the process of reloading, I see console errors in the devtools, after a "net::ERR_CONNECTION_CLOSED" error from the login url:

```
/Applications/Zulip.app/Contents/Resources/app.asar/renderer/js/preload.js:43 Uncaught ReferenceError: page_params is not defined
at HTMLDocument.document.addEventListener (/Applications/Zulip.app/Contents/Resources/app.asar/renderer/js/preload.js:43)
document.addEventListener @ /Applications/Zulip.app/Contents/Resources/app.asar/renderer/js/preload.js:43
/Applications/Zulip.app/Contents/Resources/app.asar/renderer/js/notification/index.js:24 Uncaught ReferenceError: page_params is not defined
at window.addEventListener (/Applications/Zulip.app/Contents/Resources/app.asar/renderer/js/notification/index.js:24)
```

I guess that probably should be a new issue, if one isn't already open, but I have to move on to some other things for a while. But I'll check back for updates, and thanks to those who have looked into this.
